crates/task-impls/src/da.rs

```diff
@@ -196,11 +195,7 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                     return None;
                 }
 
-                debug!(
-                    "Got a DA block with {} transactions!",
-                    proposal.data.block_payload.transaction_commitments().len()
-                );
-                let payload_commitment = proposal.data.block_payload.commit();
+                let payload_commitment = vid_commitment(proposal.data.encoded_transactions.clone());
```

Comment:
Re-computing VID commit here. Is this a DA committee member preparing to vote? If so then it's okay, otherwise we should not be computing it here.

Perhaps this computation should move down to line 219 so it's the last thing you do after all error checking and before signing the DA vote.

Why is the payload_commitment needed in line 207 for the call to `validate`? DA committee members do not check the VID commitment against anything---they compute it for themselves and sign it.

Reply:
> Is this a DA committee member preparing to vote?

Yes!

> Why is the `payload_commitment` needed in line 207 for the call to validate?

I think the VID commitment is needed as the signed data to be used to verify the signature, to make sure the DA proposal is from the expected leader. So it's not really the commitment that's being checked, but the signature of the proposal.

Reply:
> I think the VID commitment is needed as the signed data to be used to verify the signature, to make sure the DA proposal is from the expected leader. So it's not really the commitment that's being checked, but the signature of the proposal.

Ok I think I finally get it. We recently discussed that the DA leader no longer sends the commitment to the DA committee. However, the DA leader's *signature* is on the VID commitment, even though he doesn't send it. Instead each committee member re-computes the VID commitment and then checks the DA leader's signature as a signature of that re-computed commitment.

Reply:
Hmm. I feel like the signature check should come first (ie signature is on data, not commitment). Otherwise anybody can force a DA member to do the expensive VID commitment before they realize the message was not sent by the leader. This is probably something we can play around with later though, no need to change it now

Reply:
If you're signing the raw data then basically you have to hash the entire payload. So you're effectively computing two different commitments of the same payload one fast one slow.

Reply:
Yep. But one is _really_ fast...

Reply:
True. Alright I'm convinced. I'm also convinced there's no need to change it now.

Reply:
New issue #2106 . Let's make an effort to post an issue whenever we discuss something like this. I feel like these insights get forgotten in the bowels of github PR convos.